contracts/puppeteer-initia/src/contract.rs

```diff
@@ -0,0 +1,1423 @@
+        validators.len() < u16::MAX as usize,
+        StdError::generic_err("Too many validators provided")
+    );
+    deps.api.debug(&format!(
+        "WASMDEBUG: register_delegations_and_balance_query validators:{:?}",
+        validators
+    ));
+    let current_queries: Vec<u64> = puppeteer_base
+        .delegations_and_balances_query_id_chunk
+        .keys(deps.storage, None, None, Order::Ascending)
+        .collect::<StdResult<Vec<_>>>()?;
+    let messages = current_queries
+        .iter()
+        .map(|query_id| {
+            puppeteer_base
+                .delegations_and_balances_query_id_chunk
+                .remove(deps.storage, *query_id);
+            puppeteer_base.kv_queries.remove(deps.storage, *query_id);
+            NeutronMsg::remove_interchain_query(*query_id)
+        })
+        .collect::<Vec<_>>();
+
+    let mut submessages = vec![];
+    let ica = puppeteer_base.ica.get_address(deps.storage)?;
+
+    for (i, chunk) in validators
+        .chunks(config.delegations_queries_chunk_size as usize)
+        .enumerate()
+    {
+        submessages.push(SubMsg::reply_on_success(
+            new_delegations_and_balance_query_msg(
+                config.connection_id.clone(),
+                ica.clone(),
+                config.remote_denom.clone(),
+                chunk.to_vec(),
+                config.update_period,
+            )?,
+            ReplyMsg::KvDelegationsAndBalance { i: i as u16 }.to_reply_id(),
+        ));
+    }
+    Ok(Response::new()
+        .add_messages(messages)
+        .add_submessages(submessages))
+}
+
+fn register_unbonding_delegations_query(
+    deps: DepsMut<NeutronQuery>,
+    info: MessageInfo,
+    validators: Vec<String>,
+) -> ContractResult<Response<NeutronMsg>> {
+    let puppeteer_base = Puppeteer::default();
+    let config = puppeteer_base.config.load(deps.storage)?;
+    cw_ownable::assert_owner(deps.storage, &info.sender)?;
+
+    cosmwasm_std::ensure!(
+        validators.len() < u16::MAX as usize,
+        StdError::generic_err("Too many validators provided")
+    );
+
+    // TODO: this code will leave behind many registered ICQs when called again
+    //       we need to call RegisterDelegations and RegisterUnbondingDelegations together
+    //       and update existing queries
+
+    let delegator = puppeteer_base.ica.get_address(deps.storage)?;
+    let msgs = validators
+        .into_iter()
+        .enumerate()
+        .map(|(i, validator)| {
+            puppeteer_base.unbonding_delegations_reply_id_storage.save(
+                deps.storage,
+                i as u16,
+                &UnbondingDelegation {
+                    validator_address: validator.clone(),
+                    query_id: 0,
+                    unbonding_delegations: vec![],
+                    last_updated_height: 0,
+                },
+            )?;
+
+            Ok(SubMsg::reply_on_success(
+                new_register_delegator_unbonding_delegations_query_msg(
```

Comment:
As far as I know unbonding delegations is not used anymore, maybe we can deprecate this method in this version of puppeteer?